feeder/lib/feed_widget.dart

```diff
@@ -120,8 +153,172 @@ Write an article in markdown format, threading through the 3 shlokas above. Stru
 Use simple language, keep the tone devotional and practical. Make it engaging and inspiring.
 ''';
 
+class ShlokaContent {
+  final String chapterShlokaNum;
+  final String shlokaInSanskrit;
+  final String meaning;
+  final String gitabhashya;
+
+  ShlokaContent({
+    required this.chapterShlokaNum,
+    required this.shlokaInSanskrit,
+    required this.meaning,
+    required this.gitabhashya,
+  });
+}
+
+final _allBackticksPattern = RegExp(r'`[^`]+`');
+
+ShlokaContent _extractShlokaContent(String mdContent) {
+  final lines = mdContent.split('\n');
+
+  // Extract chapter-shloka number (e.g., "## 2-47")
+  String chapterShlokaNum = '';
+  for (var line in lines) {
+    if (line.startsWith('## ')) {
+      chapterShlokaNum = line.substring(3).trim();
+      break;
+    }
+  }
+
+  // Extract Sanskrit shloka from shloka-sa code block
+  String shlokaInSanskrit = '';
+  bool inShlokaSa = false;
+  for (var line in lines) {
+    if (line.contains('```shloka-sa') && !line.contains('shloka-sa-hk')) {
+      inShlokaSa = true;
+      continue;
+    }
+    if (inShlokaSa && line.trim() == '```') {
+      break;
+    }
+    if (inShlokaSa && line.trim().isNotEmpty) {
+      shlokaInSanskrit += '$line\n';
+    }
+  }
+
+  // Extract meaning (paragraph after shloka-sa-hk block, filter out Sanskrit parts)
+  String meaning = '';
+  bool afterSahk = false;
+  bool inMeaning = false;
+  for (int i = 0; i < lines.length; i++) {
+    final line = lines[i];
+
+    if (line.contains('```shloka-sa-hk')) {
+      afterSahk = true;
+      continue;
+    }
+    if (afterSahk && line.trim() == '```') {
+      inMeaning = true;
+      continue;
+    }
+    if (inMeaning) {
+      // Skip lines starting with underscore or > (these are notes/quotes)
+      if (line.trim().startsWith('_') || line.trim().startsWith('>') || line.trim().startsWith('<a name=')) {
+        break;
+      }
+      // Add non-empty lines to meaning
+      if (line.trim().isNotEmpty) {

```

Comment:
Once you encounter an empty line here, the meaning ends. Make `inMeaning` false.